python/src/GaussKronrod_doc.i.in

```diff
@@ -160,3 +160,28 @@ Parameters
 ----------
 rule : :class:`~openturns.GaussKronrodRule`
     The Gauss-Kronrod rule used in the integration algorithm."
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::GaussKronrod::GetRules
+"Get the collection of available integration rules.
+
+Returns
+-------
+rules : :class:`~openturns.Collection<~openturns.GaussKronrodRule>`
```

Comment:
same here

Reply:
@jschueller Ouch! So what can I do here?

Reply:
maybe just "sequence of int"

Reply:
@jschueller I don't understand how a `~openturns.Collection<~openturns.GaussKronrodRule>` can be seen as a sequence of int. A `~openturns.GaussKronrodRule` is a class, not an enum (a GaussKronrodRule is not a GaussKronrodPair). BTW the third argument of the constructor of GaussKronrod is a GaussKronrodRule and is mentioned this way, with no complaint from sphinx/swig and all this nightmare.

The only point I see is the fact that this kind of sequence appears for the first time in the Python interface: there is no previous declaration of OT::Collection<OT::GaussKronrodRule> in swig

Reply:
I was confused with GaussKronrodPair
we need a `%template(GaussKronrodRuleCollection) OT::Collection<OT::GaussKronrodRule>;` (in GaussKronrod.i?)
then you can use the  GaussKronrodRuleCollection type provided it must be added to base_objects.rst

Reply:
Ok I added %template(GaussKronrodRuleCollection) OT::Collection<OT::GaussKronrodRule>; in GaussKronrod.i and added GaussKronrodRuleCollection to base_objects.rst, but I got an error regarding unknown pythonTraits, so I added
```
 template <>
struct traitsPythonType< GaussKronrodRule >
{
  typedef _PySequence_ Type;
};

template <>
inline
GaussKronrodRule
convert< _PySequence_, GaussKronrodRule >(PyObject *)
{
  return GaussKronrodRule();
}
```
to PythonWrappingFunctions.hxx, but now I get an error which seems very unrelated to these changes:
```
sphinx.errors.SphinxWarning: autodoc: failed to import function 'BinomialCoefficient' from module 'openturns.SpecFunc'; the following exception was raised:
Traceback (most recent call last):
```
So close, so far...

Reply:
maybe with just https://github.com/regislebrun/openturns/pull/3 ?

Reply:
Ok thanks. Hope it will work this time ;-)

Reply:
Don't look like a success :-(

Reply:
all you need is
https://github.com/regislebrun/openturns/pull/4